server/src/connections/redis-dal.ts

```diff
@@ -0,0 +1,78 @@
+import { createClient, RedisClientType } from 'redis';
+import { REDIS_URL } from '../utils/environment-variables';
+
+interface IRedisCacheProps<T> {
+	key: string;
+	callbackFn: () => Promise<T>;
+	expirationTime: number;
+}
+
+interface IRedisSetValueProps<T> {
+	key: string;
+	value: T;
+	expirationTime: number;
+}
+
+export class RedisDAL {
+	public redisClient: RedisClientType;
+	// connect to redis
+	constructor() {
+		if (!this.redisClient) {
+			try {
+				this.redisClient = createClient({ url: REDIS_URL });
+				this.redisClient.connect().then(() => console.log('connect to Redis'));
+			} catch (error: any) {
+				// TODO - replace once AppError ready
+				throw new Error('error here');
+			}
+		}
+	}
```

Comment:
it looks like a singleton pattern `if (!this.redisClient){...})` but its not fully implemented here
as for now every time the contractor is called it will create a new connection

```suggestion
	export class RedisDAL {
	private static instance: RedisDAL;
	public redisClient: RedisClientType;

	private constructor() {
	        // connection only one time
		this.redisClient = createClient({ url: process.env.REDIS_URL });
		...
	}

	public static getInstance(): RedisDAL {
		if (!RedisDAL.instance) {
			RedisDAL.instance = new RedisDAL();
		}
		return RedisDAL.instance;
	}
```

Reply:
Nice catch, but add clearly suggestion here how to handle it differently. @NetanelSegal